src/txmempool.cpp

```diff
@@ -1090,8 +1090,9 @@ uint64_t CTxMemPool::CalculateDescendantMaximum(txiter entry) const {
     return maximum;
 }
 
-void CTxMemPool::GetTransactionAncestry(const uint256& txid, size_t& ancestors, size_t& descendants) const {
-    LOCK(cs);
+void CTxMemPool::GetTransactionAncestry(const uint256& txid, size_t& ancestors, size_t& descendants) const
+{
+    AssertLockHeld(cs);
```

Comment:
Again, I don't think this change is necessary. `GetTransactionAncestry()` is only called with the lock held in the tests, which can be updated.

Reply:
Right, I also saw that. I guess @hebasto preferred this approach because the changes to tests would be too big and ugly? (if we mandate that `GetTransactionAncestry()` will lock the mutex itself and the callers should not be holding it)

Reply:
@vasild Correct.